pyproject.toml

```diff
@@ -194,23 +198,75 @@ SQLAlchemy = ">=1.4.0"
 pymysql = "^1.1.0"
 pypdf2 = "^3.0.1"
 pydoc-markdown = "^4.8.2"
-connectorx="0.3.2"
-dbt-core=">=1.2.0"
-dbt-duckdb=">=1.2.0"
+connectorx = "0.3.2"
+dbt-core = ">=1.2.0"
+dbt-duckdb = ">=1.2.0"
 pymongo = ">=4.3.3"
 pandas = ">2"
 
-[tool.black] # https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#configuration-via-a-file
+[tool.ruff]
 line-length = 100
-preview = true
+src = ["dlt", "tests"]
+
+[tool.ruff.format]
+exclude = ["*.pyi"]
+
+[tool.ruff.lint]
+select = ["A", "B", "Q", "F", "S", "UP"]
+exclude = ["tests/reflection/module_cases/**/*.py"]
+
+# Allow ruff to fix the following cases
+fixable = [
+    "UP008", # Use super() instead of super(__class__, self)
+    "UP015", # Unnecessary open mode parameters
+    "UP028", # Replace yield over for loop with yield from
+    "UP031", # Use format specifiers instead of percent format
+    "UP032", # Use f-string instead of `format` call
+]
 
-[tool.isort] # https://pycqa.github.io/isort/docs/configuration/options.html
-color_output = true
-line_length = 100
-profile = "black"
-src_paths = ["dlt"]
-multi_line_output = 3
+[tool.ruff.lint.per-file-ignores]
+"dlt/**/*.py" = [
+    "F401",  # allow unused imports
+    "S101",  # allow assert
+    "S608",  # Possible SQL injection vector through string-based query construction
```

Comment:
@sh-rp @rudolfix should keep this disabled or should we sanitize certain SQL queries in the future?